snakesist/exist_client.py

```diff
@@ -50,24 +51,15 @@ def __init__(
         self._exist_client = exist_client
 
         if query_result:
-            self._abs_resource_id, self._node_id, self.node = query_result
+            self._abs_resource_id, self._node_id, self.path, self.node = query_result
```

Comment:
if that `path` is the same as the `collection_path` that comes up later, i would name it exactly like that. in an application there's often a lot of different paths being used and more specific names certainly help to avoid confusion stemming from that.

addendum: that actually helped me realize the answer to a question i had: a document has both an absolute id and a collection path is is located *in*, correct?

Reply:
good point about the naming, i'll rename them all to `collection_path`.

the document has both an absolute id and a collection path pointing to it, that's correct. i initially hoped to be able to do all operations using the ids, but it seems that there is no xquery function which allows you to update or delete a resource by its id, so those two operations are now done by using post and delete requests, which require the path to the resource. might actually be better, it feels like the more RESTful way to do it...

Reply:
i see. flat identifiers have their charmes though and one might point the exist folks on that inconsistency.

Reply:
i'm also worried a bit about the stability of the eXist functionality using absolute ids. while running the test suite against multiple eXist versions i discovered that the xquery function which you can use to retrieve a resource by its absolute id [is broken in 5.0.0](https://github.com/eXist-db/exist/issues/3031).

i'll see if i can find out what's the optimal and most stable way to identify resources from the exist perspective and if they plan on supporting and extending the absolute id functionality.